### PR TITLE
LokiContextUi: Add loading indicator

### PR DESCRIPTION
**Pull request by svennergr**
**What is this feature?**

Adds a loading indicator to the `LokiContextUi`. While the user is adding or removing labels a loading indicator is displayed.

**Special notes for your reviewer**:

This can be tested by running a Grafana with the `logsContextDatasourceUi` feature toggle turned on. Just make a Loki query containing a line filter, open show context and de-/select filters.

### CODE REVIEW

File: public/app/plugins/datasource/loki/components/LokiContextUi.tsx [hunk 1]
```diff
@@ -153,6 +169,7 @@ export function LokiContextUi(props: LokiContextUiProps) {
           />
         </Tooltip>{' '}
         Select labels to be included in the context query:
+        <LoadingPlaceholder text="" className={`${styles.loadingPlaceholder} ${loading ? '' : styles.hidden}`} />
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/components/LokiContextUi.tsx:172]: Wouldn't it be better to solve the height of div in `styles.textWrapper`  and then just do `{loading ? <LoadingPlaceholder className={styles.loadingPlaceholder}/> : null}`. Or set positioning in loadingPlaceholder to `absolute` with `top: 0` `right: spacing.1`.

Comment (svennergr) [reply public/app/plugins/datasource/loki/components/LokiContextUi.tsx:172]: I just went with this approach because 
1. this is "easier" to implement
2. this is fast/more efficient in the browser. This is quite useless here, because there is only one component, but `{loading ? <LoadingPlaceholder className={styles.loadingPlaceholder}/> : null}` is basically mounting a component and unmounting it to the DOM, while my solution is just changing a CSS class.

Do you think my approach is okay, or should I do `{loading ? <LoadingPlaceholder className={styles.loadingPlaceholder}/> : null}`?

File: public/app/plugins/datasource/loki/datasource.ts [hunk 2]
```diff
@@ -789,53 +790,60 @@ export class LokiDatasource
...
-
-          const parserContextFilters = contextFilters.filter((filter) => filter.fromParser && filter.enabled);
-          if (parserContextFilters.length) {
-            // we should also filter for labels from parsers, let's find the right parser
-            if (origQuery) {
-              const parser = getParserFromQuery((origQuery as LokiQuery).expr);
-              if (parser) {
-                expr = addParserToQuery(expr, parser);
-              }
+    const updateFilter = (contextFilters: ContextFilter[]) => {
```
Comment (ivanahuckova) [inline public/app/plugins/datasource/loki/datasource.ts:793]: @svennergr could you please share why is this change needed in relation to loading indicator. I am not sure I understand.

File: public/app/plugins/datasource/loki/components/LokiContextUi.tsx (hunk 1 above)
Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/components/LokiContextUi.tsx:172]: 👍

File: public/app/plugins/datasource/loki/datasource.ts (hunk 2 above)
Comment (svennergr) [reply public/app/plugins/datasource/loki/datasource.ts:793]: Sure. Previously the whole component `LokiContextUi` was rerendered on each call. That resulted in different references being generated for the `onClose` property. That resulted in this `onEffect` being called too many times and too early: https://github.com/grafana/grafana/pull/64167/files/2feed00b8817c536231c706617eb0588a0762b55#diff-482a82afb5742195c02896eecc365131b6283b498d80446168cb7daf64286f3cL90
Which then resulted in an too early `clearTimeout` call.

Comment (ivanahuckova) [reply public/app/plugins/datasource/loki/datasource.ts:793]: gotcha, thanks for context!

**Review by ivanahuckova — APPROVED** [review]
LGTM
